### PR TITLE
update auto redirect; add name regex check

### PR DESCRIPTION
**Pull request by kunmingg**
- update auto redirect target image
- check deploy name input format

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/4082)

### CODE REVIEW

File: components/gcp-click-to-deploy/src/DeployForm.tsx [hunk 1]
```diff
@@ -784,10 +785,10 @@ export default class DeployForm extends React.Component<any, DeployFormState> {
                     if (ready_test != null) {
                         setTimeout(() => {
                             // We rotate on image addresses of v0.4 and v0.5+ t to support both v0.4 and v0.5+
-                            if (ready_test.src.includes('hub/logo')) {
+                            if (ready_test.src.includes('favicon')) {
```
Comment (abhi-g) [inline components/gcp-click-to-deploy/src/DeployForm.tsx]: do we still need the if else test? Looks like with the line 774 update, favicon is always going to be in the src?

File: components/gcp-click-to-deploy/src/DeployForm.tsx [hunk 2]
```diff
@@ -784,10 +785,10 @@ export default class DeployForm extends React.Component<any, DeployFormState> {
                     if (ready_test != null) {
                         setTimeout(() => {
                             // We rotate on image addresses of v0.4 and v0.5+ t to support both v0.4 and v0.5+
```
Comment (abhi-g) [inline components/gcp-click-to-deploy/src/DeployForm.tsx]: may be remove the comment if we end up getting rid of the if/else block

File: components/gcp-click-to-deploy/src/DeployForm.tsx [hunk 3]
```diff
@@ -884,6 +885,15 @@ export default class DeployForm extends React.Component<any, DeployFormState> {
             });
             throw err;
         }
+        const filtered = this.state[deploymentNameKey].match(nameformat);
+        if (!(filtered && this.state[deploymentNameKey] === filtered[0])) {
+            this.setState({
+                dialogAsCode: false,
+                dialogBody: 'Deployment name need to match rgular expression: [a-z]([-a-z0-9]*[a-z0-9])?',
```
Comment (abhi-g) [inline components/gcp-click-to-deploy/src/DeployForm.tsx]: s/rgular/regular/

more importantly, should we make this more human readble. ie., name needs to be 25 chars or less of lowercase alphanumeric characters and hyphens.

File: components/gcp-click-to-deploy/src/DeployForm.tsx (hunk 1 above)
Comment (kunmingg) [reply components/gcp-click-to-deploy/src/DeployForm.tsx]: We might have to iterate for v0.5 compatibility, v0.6 and v0.5 have different image location

File: components/gcp-click-to-deploy/src/DeployForm.tsx (hunk 3 above)
Comment (kunmingg) [reply components/gcp-click-to-deploy/src/DeployForm.tsx]: updated

**Pull request merged**